pkg/service/replay/replay.go

```diff
@@ -478,6 +487,35 @@ func (r *replayer) RunTestSet(ctx context.Context, testSetID string, testRunID s
 		return models.TestSetStatusInternalErr, fmt.Errorf("failed to insert report")
 	}
 
+	// remove the unused mocks by the test cases of a testset
+	if r.config.Test.ClearUnusedMocks {
+
+		// fetch the consumed mocks by the testcases of the testset
+		consumedMocks, err := r.instrumentation.GetConsumedMocks(runTestSetCtx, appID)
+		if err != nil {
+			utils.LogError(r.logger, err, "failed to get consumed mocks", zap.Any("for test-set", testSetID))
+		}
+		r.logger.Debug("consumed mocks from the completed testset", zap.Any("for test-set", testSetID), zap.Any("consumed mocks", consumedMocks))
+		// if the mock is not consumed by the testset then it is unused
+		unusedMocks := map[string]bool{}
+		for _, filteredMock := range filteredMocks {
+			if _, ok := consumedMocks[filteredMock.Name]; !ok {
+				unusedMocks[filteredMock.Name] = false
```

Comment:
the bool is true for config mocks only. 
Added a comment in the function implementation: [here](https://github.com/keploy/keploy/blob/f5ade3ca1360b094a7256356f3b5ee262e03ac16/pkg/platform/yaml/mockdb/db.go#L41)

Reply:
even in that case you can see there it is config or not from the mocks retrieved from yaml

Reply:
i have used maps because the search complexity will O(1), this will help in easily check that the mock should be deleted or not in the deleteMocks. Since, currently for yaml files there are not the Delete helper function which can go and delete that particular yml doc. So we have to read, iterate and exclude the mocks with name present in the map.

Reply:
> why do we need it, you want to delete only tcs mocks..?

No, I am sending this as a map for better approach to search in the mock names, And in future if there is any any possibilty for config/tcs mocks. This can easily handle that